### PR TITLE
Add standard gitignore

### PR DESCRIPTION
**Pull request by craig-mulligan**
Copy of https://github.com/github/gitignore/blob/master/Node.gitignore

Change-Type: patch

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+# Compiled binary addons (http://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
+jspm_packages/
+
+# Typescript v1 declaration files
+typings/
```
Comment (CameronDiver) [inline .gitignore]: I don't think that this should be in the .gitignore, I use this directory to put in custom typings for modules which don't have any, and that seems to be the standard.

Comment (hedss) [inline .gitignore]: :+1: on not ignoring `typings`, as per the TypeScript guide, any code that has no typings and is being used needs defining in here, and you want to include that directory in the repo.

Comment (craig-mulligan) [inline .gitignore]: Sweet, any opinion on ignoring `/build`? I'd prefer to, but I know some like to keep it can be easily installed from github.

Comment (CameronDiver) [inline .gitignore]: I think that we tend to keep it around so that a user doesn't have to install transpilers etc.

Comment (hedss) [inline .gitignore]: Yeah, `build` needs to exist so that the NPM module can be used `as is`.

**Pull request merged**